coreax/kernels/scalar_valued.py

```diff
@@ -30,28 +30,42 @@ class LinearKernel(ScalarValuedKernel):
     r"""
     Define a linear kernel.
 
-    Given :math:`\rho`=`'output_scale` and :math:`c =`'constant',  the linear kernel is
+    Given :math:`\rho`=` ``output_scale``, :math:`a=` ``constant`` and
+    :math: `b =` ``offset``,  the linear kernel is
     defined as :math:`k: \mathbb{R}^d\times \mathbb{R}^d \to \mathbb{R}`,
-    :math:`k(x, y) = \rho x^Ty + c`.
+    :math:`k(x, y) = a + \rho (x-b)^T(y-b)`.
 
     :param output_scale: Kernel normalisation constant, :math:`\rho`
-    :param constant: Additive constant, :math:`c`
+    :param constant: Additive constant, :math:`a`, must be positive
+    :param offset: Offset term, :math:`b`
     """
 
     output_scale: float = eqx.field(default=1.0, converter=float)
     constant: float = eqx.field(default=0.0, converter=float)
+    offset: float = eqx.field(default=0.0, converter=float)
+
+    def __check_init__(self):
+        """Check that the additive constant is not negative."""
+        if self.constant < 0:
```

Comment:
Yes, this would need adding to every kernel in the codebase though. I guess we have been assuming up to this point people choose sensible parameters for these things.

Reply:
I could add `__check_init__`'s to every kernel or remove the one I added to stay consistent?

Reply:
See #718